### PR TITLE
FLUT-869447 - [Others] Added user guidelines for behavior public methods in development

### PR DESCRIPTION
**Pull request by natrayansf**
## Description ##

Added user guidelines for trackball and crosshair behavior public methods

### CODE REVIEW

File: Flutter/cartesian-charts/methods.md [hunk 1]
```diff
@@ -459,6 +459,167 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
 {% endhighlight %}
 {% endtabs %}
 
+### Events in trackballBehavior
+
+Integrated the following methods for handling pointer and gesture events and allowing customizations for various pointer events such as long-press, tap, double-tap, pointer enter, and exit.
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:464]: Provided

File: Flutter/cartesian-charts/methods.md [hunk 2]
```diff
@@ -459,6 +459,167 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+
+* `handleEvent` - Specifies to customize the necessary pointer events.
+* `handleLongPressStart` - Specifies to customize the pointer when a long-press begins.
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:467]: Specifies to customize the pointer event

File: Flutter/cartesian-charts/methods.md [hunk 3]
```diff
@@ -459,6 +459,167 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+* `handleTapDown` - Specifies to customize the pointer when a tap has contacted the screen once.
+* `handleTapUp` - Specifies to customize the pointer when it has stopped contacting the screen after a tap.
+* `handleDoubleTap` - Specifies to customize the pointer when a tap has contacted the screen twice.
+* `handlePointerEnter` -Specifies to customize the pointer when the mouse enter on the screen.
+* `handlePointerExit` - Specifies to customize the pointer when the mouse exit on the screen.
+
+{% tabs %}
+{% highlight dart %} 
+
+    class CustomTrackballEvent extends TrackballBehavior {
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:479]: Add code snippet for custom trackball code in chart

File: Flutter/cartesian-charts/methods.md [hunk 4]
```diff
@@ -459,6 +459,167 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+      @override
+      bool get enable => true;
+
+      @override
+      ActivationMode get activationMode => super.activationMode;
+
+      // Stops the behavior of mouse pointer move, cancel, hover and up events.
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:486]: Disable the behavior of pointer move,

**Review by LavanyaGowtham2021 — CHANGES_REQUESTED** [review]

File: Flutter/cartesian-charts/methods.md [hunk 5]
```diff
@@ -653,6 +815,338 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+### DrawVerticalAxisLine method in crosshairBehavior
+
+To customize the stroke drawing and styling of vertical crosshair line.
+
+* `context` - Holds the painting context details necessary for drawing on the canvas.
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:876]: Specifies the painting context.

File: Flutter/cartesian-charts/methods.md [hunk 6]
```diff
@@ -653,6 +815,338 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+* `offset` - It specifies the crosshair position.
+* `dashArray` - By default, the dashArray is applied based on the [`lineDashArray`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/CrosshairBehavior/lineDashArray.html) property which is available in the [`crosshairBehavior`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/charts/CrosshairBehavior/CrosshairBehavior.html) class.
```
Comment (LavanyaGowtham2021) [inline Flutter/cartesian-charts/methods.md:878]: Specifies the crosshair line dashArray.

Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md]: Used this "Specifies the crosshair line dashArray".

File: Flutter/cartesian-charts/methods.md [hunk 7]
```diff
@@ -653,6 +815,338 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+
+{% endhighlight %}
+{% endtabs %}
+
+### DrawVerticalAxisLine method in crosshairBehavior
+
+To customize the stroke drawing and styling of vertical crosshair line.
+
+* `context` - Holds the painting context details necessary for drawing on the canvas.
+* `offset` - It specifies the crosshair position.
```
[the remark this answers is not in the document]
Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md]: Used this content "Specifies the crosshair position."

File: Flutter/cartesian-charts/methods.md (hunk 5 above)
Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md]: Used this "Specifies the painting context" content.

File: Flutter/cartesian-charts/methods.md [hunk 8]
```diff
@@ -459,6 +459,168 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+
+{% tabs %}
+{% highlight dart %} 
+
+    SfCartesianChart(
+      trackballBehavior: CustomTrackballEvent(),
+      // Add series.
+    );
+    
+    class CustomTrackballEvent extends TrackballBehavior {
```
[the remark this answers is not in the document]
Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md]: Naming changed "CustomTrackball"

File: Flutter/cartesian-charts/methods.md [hunk 9]
```diff
@@ -459,6 +459,166 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+* `handleLongPressStart` - Specifies to customize the pointer event when a long-press begins.
+* `handleLongPressMoveUpdate` - Specifies to customize the pointer event during movement after a long-press.
+* `handleLongPressEnd` - Specifies to customize the pointer event when the pointer stops contacting the screen after a long-press.
+* `handleTapDown` - Specifies to customize the pointer event when a tap has contacted the screen once.
+* `handleTapUp` - Specifies to customize the pointer event when it has stopped contacting the screen after a tap.
+* `handleDoubleTap` - Specifies to customize the pointer event when a tap has contacted the screen twice.
+* `handlePointerEnter` -Specifies to customize the pointer event when the mouse enter on the screen.
+* `handlePointerExit` - Specifies to customize the pointer event when the mouse exit on the screen.
+
+This following code snippet defines how to enable/disable specific pointer events and customize to show trackball in double tap and disabled trackball showing in pointer enter and exit.
```
Comment (M-A32) [inline Flutter/cartesian-charts/methods.md:476]: This following code sample defines how to enable or disable specific pointer events and customize to show trackball in double tap and disabled trackball showing in pointer enter and exit.

File: Flutter/cartesian-charts/methods.md [hunk 10]
```diff
@@ -653,6 +813,336 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+* `handleLongPressStart` - Specifies to customize the pointer event when a long-press begins.
+* `handleLongPressMoveUpdate` - Specifies to customize the pointer event during movement after a long-press.
+* `handleLongPressEnd` - Specifies to customize the pointer event when the pointer stops contacting the screen after a long-press.
+* `handleTapDown` - Specifies to customize the pointer event when a tap has contacted the screen once.
+* `handleTapUp` - Specifies to customize the pointer event when it has stopped contacting the screen after a tap.
+* `handleDoubleTap` - Specifies to customize the pointer event when a tap has contacted the screen twice.
+* `handlePointerEnter` -Specifies to customize the pointer event when the mouse enter on the screen.
+* `handlePointerExit` - Specifies to customize the pointer event when the mouse exit on the screen.
+
+This following code snippet defines how to enable/disable specific pointer events and customize to show crosshair in double tap and disabled crosshair showing in pointer enter and exit.
```
Comment (M-A32) [inline Flutter/cartesian-charts/methods.md:830]: This following code sample defines how to enable or disable specific pointer events and customize to show crosshair in double tap and disabled crosshair showing in pointer enter and exit.

File: Flutter/cartesian-charts/methods.md [hunk 11]
```diff
@@ -653,6 +813,336 @@ The [`hide`](https://pub.dev/documentation/syncfusion_flutter_charts/latest/char
...
+{% endhighlight %}
+{% endtabs %}
+
+![Crosshair Horizonatl Axis Tooltip](images\trackball-crosshair/custom_crosshair_horizontal_tooltip.png)
+
+### DrawVerticalAxisTooltip method in crosshairBehavior
+
+To customize the tooltip label, position, and style for vertical axis tooltip.
+
+For further reference, please consult the drawHorizontalAxisTooltip code snippet.
```
Comment (M-A32) [inline Flutter/cartesian-charts/methods.md:991]: For further reference, please consult the drawHorizontalAxisTooltip code sample.

**Review by M-A32 — CHANGES_REQUESTED** [review]
I have completed the review.

File: Flutter/cartesian-charts/methods.md (hunk 4 above)
Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md]: Used this "Disable the behavior of the pointer move"

File: Flutter/cartesian-charts/methods.md (hunk 10 above)
Comment (natrayansf) [reply Flutter/cartesian-charts/methods.md:830]: Used the following content for crosshair "This following code sample defines how to enable or disable specific pointer events and customize to show crosshair in double tap and disabled crosshair showing in pointer enter and exit."

**Review by LavanyaGowtham2021 — APPROVED** [review]
Fine

**Pull request merged**